integration/sawtooth_integration/tests/test_identity.py

```diff
@@ -0,0 +1,399 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------
+
+import enum
+import subprocess
+import unittest
+from uuid import uuid4
+
+import cbor
+
+from sawtooth_processor_test.message_factory import MessageFactory
+
+from sawtooth_integration.tests.integration_tools import RestClient
+from sawtooth_integration.tests.integration_tools import wait_for_rest_apis
+
+from sawtooth_sdk.protobuf import batch_pb2
+
+
+class TestTransactorPermissioning(unittest.TestCase):
+
+    def setUp(self):
+        self.REST_ENDPOINT = "rest-api:8080"
+        wait_for_rest_apis([self.REST_ENDPOINT])
+        self.alice = Transactor('alice', self.REST_ENDPOINT)
+        self.bob = Transactor('bob', self.REST_ENDPOINT)
+        self.carol = Transactor('carol', self.REST_ENDPOINT)
+        self.dave = Transactor('dave', self.REST_ENDPOINT)
+
+        self.chuck = Transactor('chuck', self.REST_ENDPOINT)
+        self.mallory = Transactor('mallory', self.REST_ENDPOINT)
+
+        self.walter = Transactor('walter', self.REST_ENDPOINT)
+
+    def test_transactor_permissioning(self):
+        """Test the transactor permissioning system using the Identity
+        transaction family.
+
+        Notes:
+            The test works from general to specific on the transactor
+            permissioning settings, starting with 'transactor', then
+            'transactor.batch_signer', then 'transactor.transaction_signer',
+            and then finally 'transactor.transaction_signer.intkey' and
+            'transaction.transaction_signer.xo'. For each subsection,
+            the test allows some transactors and denies others and asserts
+            that each either is or is not able to send transactions.
+
+            From local configuration, Chuck and Mallory are denied from
+            being transactors, Dave is denied from being a batch_signer,
+            Carol is denied from sending XO transactions, while Walter, Bob,
+            and Alice have all transactor permissions and no explicit denials
+            from more specific permissions.
+
+        """
+
+        #
+        # transactor subsection
+        #
+        # From local configuration Dave is denied from being a batch_signer,
+        # and Chuck and Mallory are denied from all transactor permissions.
+
+        self.walter.set_pubkey_for_role(
+            'deny_bob_allow_alice_walter',
+            'transactor',
+            permit_keys=[self.alice.public_key, self.walter.public_key],
+            deny_keys=[self.bob.public_key])
+
+        for t, f in [(self.alice, Families.INTKEY),
```

Comment:
I'd prefer if `t` and `f` were written out in full (the list pairs can be indented if needed).

Reply:
Actually, it might be better to have the assertions expect a variable number of pairs, so the test body wouldn't have the loops at all. Then the assertions would look like
```
self.assert_able_to_send(
    (self.alice, Families.INTKEY),
    (self.walter, Families.INTKEY),
    ...,
)

Reply:
Updating.